File: src/main/java/com/nndmove/app/service/dto/HistoryDTO.java
package com.nndmove.app.service.dto;

import jakarta.validation.constraints.*;
import java.io.Serializable;
import java.util.Objects;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO for the {@link com.nndmove.app.domain.History} entity.
 */
@Getter
@Setter
@NoArgsConstructor
@SuppressWarnings("common-java:DuplicatedBlocks")
public class HistoryDTO implements Serializable {

    private Long id;

    @NotNull
    private Integer part;

    @NotNull
    private Integer stopTime;

    private UserDTO user;

    private MovieDTO movie;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof HistoryDTO)) {
            return false;
        }

        HistoryDTO historyDTO = (HistoryDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, historyDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "HistoryDTO{" +
            "id=" + getId() +
            ", part=" + getPart() +
            ", stopTime=" + getStopTime() +
            ", user=" + getUser() +
            ", movie=" + getMovie() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/service/dto/HistoryDTO.kt
package com.nndmove.app.service.dto

import jakarta.validation.constraints.NotNull
import java.io.Serializable
import java.util.*

/**
 * A DTO for the [com.nndmove.app.domain.History] entity.
 */
class HistoryDTO : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var part: @NotNull Int? = null

    @JvmField
    var stopTime: @NotNull Int? = null

    @JvmField
    var user: UserDTO? = null

    @JvmField
    var movie: MovieDTO? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is HistoryDTO) {
            return false
        }

        if (this.id == null) {
            return false
        }
        return this.id == o.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }

    // prettier-ignore
    override fun toString(): String {
        return "HistoryDTO{" +
                "id=" + id +
                ", part=" + part +
                ", stopTime=" + stopTime +
                ", user=" + user +
                ", movie=" + movie +
                "}"
    }
}
